ui/src/components/QRWidget.tsx

```diff
@@ -0,0 +1,56 @@
+import { useMemo } from 'react';
+import cn from 'classnames';
+import QRCode from 'react-qr-code';
+import { useCopy } from '@/logic/utils';
+import CopyIcon from '@/components/icons/CopyIcon';
+import CheckIcon from './icons/CheckIcon';
+import ShareIcon from './icons/ShareIcon';
+
+export default function QRWidget({
+  link,
+  className,
+  navigatorTitle,
+}: {
+  link: string;
+  className?: string;
+  navigatorTitle: string;
+}) {
+  const url = useMemo(() => new URL(link), [link]);
+  const { didCopy, doCopy } = useCopy(link);
+  const displayURL = url.hostname + url.pathname;
+
+  const handleCopy = () => {
+    if (navigator.share !== undefined) {
+      navigator.share({
+        title: navigatorTitle,
+        url: link,
+      });
+    } else {
+      doCopy();
+    }
+  };
+
+  return (
+    <div
+      className={cn(
+        'flex h-full w-full flex-col rounded-xl bg-blue-200',
+        className
+      )}
+      onClick={handleCopy}
+    >
+      <div className="flex flex-1 items-center items-center justify-center py-8">
+        <QRCode value={link} size={200} fgColor="#bfdbfe" bgColor="#3b82f6" />
```

Comment:
Does this generate a SVG or a base-64 inline PNG? It needs to be a SVG to play nice with Hosting's remote image policy (LAND-1045).